superset-frontend/src/views/CRUD/data/query/QueryList.tsx

```diff
@@ -80,17 +86,16 @@ const StyledPopoverItem = styled.div`
   color: ${({ theme }) => theme.colors.grayscale.dark2};
 `;
 
-const StatusIcon = styled(Icon, {
-  shouldForwardProp: prop => prop !== 'status',
-})<{ status: string }>`
-  color: ${({ status, theme }) => {
-    if (status === 'success') return theme.colors.success.base;
-    if (status === 'failed') return theme.colors.error.base;
-    if (status === 'running') return theme.colors.primary.base;
-    if (status === 'offline') return theme.colors.grayscale.light1;
+const getColor = (theme: SupersetTheme, status: string) => {
+  if (status === 'success') return theme.colors.success.base;
+  if (status === 'failed') return theme.colors.error.base;
+  if (status === 'running') return theme.colors.primary.base;
+  if (status === 'offline') return theme.colors.grayscale.light1;
+  return theme.colors.grayscale.base;
+};
 
-    return theme.colors.grayscale.base;
-  }};
+const StatusIcon = (theme: SupersetTheme, status: string) => css`
```

Comment:
I think the best approach here is to use the `iconColor` prop directly.

`<Icons.Check iconColor={theme.colors.primary.base} />`

With that, you can delete `getColor` and `StatusIcon`.

Reply:
Deleting the function will make getting the status color of the icons a little less dynamic since they are being used more than once in a couple of the icons.

Reply:
This makes sense. One of the good reasons to use `styled` is for this sort of reusability... I think we would still need a `getColor` method to choose the right color for each of these `iconColor` implementations anyway, unless I'm mistaken.